### PR TITLE
Ensure cert exists

### PR DESCRIPTION
**Pull request by mhahn**
example:

```
pre_build:
  - path: stacker.hooks.iam.ensure_server_cert_exists
    required: true
    args:
      cert_name: ${vault_elb_cert_name}
  - path: stacker.hooks.iam.ensure_server_cert_exists
    required: true
    args:
      cert_name: ${empire_controller_cert_name}
```

makes it easy to use private certs for different environments

### CODE REVIEW

File: stacker/hooks/iam.py [hunk 1]
```diff
@@ -30,3 +34,82 @@ def create_ecs_service_role(region, namespace, mappings, parameters,
...
+
+        paths = {
+            'certificate': None,
+            'private_key': None,
+            'chain': None,
+        }
+
+        for key in paths.keys():
+            path = raw_input('Path to %s (skip): ' % (key,))
+            if path == 'skip':
```
Comment (phobologic) [inline stacker/hooks/iam.py]: This makes it look like skip is the default, but that doesn't seem to be the case, or am I confused?

File: stacker/hooks/iam.py [hunk 2]
```diff
@@ -30,3 +34,82 @@ def create_ecs_service_role(region, namespace, mappings, parameters,
...
+    except BotoServerError:
+        upload = raw_input(
+            'Certificate "%s" wasn\'t found. Upload it now? (yes/no) ' % (
+                cert_name,
+            )
+        )
+        if upload != 'yes':
+            return False
+
+        paths = {
```
Comment (phobologic) [inline stacker/hooks/iam.py]: worth maybe making this something you can have in the hook definition?  So you can define the paths to the certs on the local FS?

File: stacker/hooks/iam.py (hunk 1 above)
Comment (mhahn) [inline stacker/hooks/iam.py]: skip isn't the default, if you explicitly write "skip" we'll skip. if you don't enter anything, you'll get a path doesn't exist error

File: stacker/hooks/iam.py (hunk 2 above)
Comment (mhahn) [inline stacker/hooks/iam.py]: good idea

**Comment by mhahn** [comment]
@phobologic comments addressed

File: stacker/hooks/iam.py (hunk 1 above)
Comment (phobologic) [inline stacker/hooks/iam.py]: `(skip)` says to me `this is the default` - maybe be a bit more explicit here?

Comment (mhahn) [inline stacker/hooks/iam.py]: ok, skip is the default now

**Pull request merged**